test/unit/dataset_spec.js

```diff
@@ -80,6 +80,10 @@ describe('Dataset', function() {
       expect(this.dataset.getRoot()).toContainText('empty');
     });
 
+    it('should throw an error if suggestions is not an array', function() {
+      expect(this.source.and.callFake(fakeGetNonArrayResults)).toThrow();
```

Comment:
you can assert some of the text of the array. I forgot if this is Jest or not, but if it is: `toThrowErrorMatchingSnapshot()`